static/js/formatters-internal.js

```diff
@@ -554,6 +554,12 @@ export function highlightField(fieldValue, matchedSubstrings = []) {
       highlightedString.substr(offset + length);
   });
 
+  /**
+   * Use regex with lookahead syntax to match all tags that's not <mark> or </mark> tag,
+   * and replace angle brackets with '&lt;' and '&gt;' to properly display them as non-html code.
+   * If matched, preserve the content ($2) inside but replace the surrounding angle brackets.
+   */
+  highlightedString = highlightedString.replace(/(<)((?!\/?mark)[^>]*)(>)/g,'&lt;$2&gt;');
```

Comment:
If there are multiple angle brackets, then they will still be left in the string. (e.g. "<<<script>>>" becomes `&lt;<<\script&gt;>>`). I'm wondering if it would be simpler to do a full escape of `<`, `>`, and `&`, and then we unescape just the mark tags

Reply:
Alternatively we could indicate the marks with some other syntax, do a full HTML escape, and then transform the mark tags into HTML

Reply:
I was thinking the replacement should happen before the forEach loop. That way we don't have to rely on any kind of regex, or any issues with somebody including `<mark>` tags in the original string

Reply:
I'm a little wary of trying to accomplish this via a complex regex. We've been bitten by that before. As Connor noted, it's hard to get it right. Instead of trying to roll our own here, maybe we can use an existing library for escaping the HTML (such as https://www.npmjs.com/package/escape-html). I like Connor's idea of marking the highlights with some other (non-HTML) syntax, doing the escaping, then adding in the mark tags.

Reply:
Specifically I was thinking
1. slice up the string into substrings
2. replace all angle brackets in the substrings
3. wrap all substrings (except for the first and last) with mark tags

Reply:
I had the replacement before the for loop initially but that messes up the indexes. Connor's suggestion makes sense, I will try out the library and add in the mark tag afterward